src/VSSDK.BuildTools/Xamarin.VSSDK.BuildTools.DeployVsixExtensionFiles.targets

```diff
@@ -0,0 +1,114 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- This target improves the built-in DeployVsixExtensionFiles target by copying (instead of unzipping) the VSIX source items 
+       without requiring to set the CreateVsixContainer property. So this is much faster because:
+       
+       1. It does not zip the VSIX source items into the .vsix container file
+       2. It avoids to unzip its content into the VsixDeploymentPath
+       3. It uses Inputs/Outputs to avoid copying unchanged files
+  -->
+
+  <PropertyGroup>
+  </PropertyGroup>
+
+  <!--
+    =================================================================================
+                          Overrrides Deploy VSIX Extension Files
+    =================================================================================
+  -->
+
+  <PropertyGroup>
+    <DeployVsixExtensionFilesDependsOn>
+      $(DeployVsixExtensionFilesDependsOn);
+      _UninstallVsixExtension;
+      _GetVSIXSourceItemWithTargetPath;
+      _CopyVsixExtensionFiles;
+      _EnableVsixExtension
+    </DeployVsixExtensionFilesDependsOn>
+  </PropertyGroup>
+
+  <Target Name="DeployVsixExtensionFiles" Condition="'$(DeployExtension)'=='true'" DependsOnTargets="$(DeployVsixExtensionFilesDependsOn)" />
+
+  <Target Name="_UninstallVsixExtension"
+          Condition="'$(ExistingDeploymentPath)' != '$(VsixDeploymentPath)' AND '$(ExistingDeploymentPath)' != '' AND '$(AllUsersExtension)' != 'true'">
+
+    <!-- Uninstall any existing extension with the same identifier with a different deployment path.
+             We don't always call this because of the way Uninstall is implemented in Extension Manager.
+             An uninstall operation just marks the directory for deletion in the registry and does the actual
+             delete on the next instantiation of the Extension Manager service object. In our case, this happens
+             at the same time (or just after) we are copying files to that directory. Therefore, we should only
+             call uninstall if the existing installation directory is different than the one we're deploying files
+             to. -->
+    <UninstallExtension
+        VsixIdentifier="$(VsixID)"
+        RootSuffix="$(VSSDKTargetPlatformRegRootSuffix)"
+        FailIfNotInstalled="false" />
+
+    <Message Text="VsixID = $(VsixID)" />
+    <Message Text="VsixVersion = $(VsixVersion)" />
+    <Message Text="VsixDeploymentPath = $(VsixDeplsoymentPath)" />
+
+  </Target>
+
+  <Target Name="_GetVSIXSourceItemWithTargetPath" DependsOnTargets="GetVsixSourceItems">
+
+    <ItemGroup>
+      <VSIXSourceItemWithTargetPath Include="@(VSIXSourceItem->Distinct())" Condition="'%(VSIXSourceItem.TargetPath)' != ''" />
+      <VSIXSourceItemWithTargetPath Include="@(VSIXSourceItem->Distinct())" Condition="'%(VSIXSourceItem.TargetPath)' == ''">
+        <TargetPath>%(Filename)%(Extension)</TargetPath>
```

Comment:
Should this be `%(RecursiveDir)\%(Filename)%(Extension)`? This would cover cases where users add items with, say `Include="Content\**\*.*"`, for example. Maybe `%(RelativeDir)` is even more generic as it would also accomodate fixed relative dirs too?